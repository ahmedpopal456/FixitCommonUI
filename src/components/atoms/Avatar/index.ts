import { AvatarBadge } from './Avatar';
export { AvatarBadge};