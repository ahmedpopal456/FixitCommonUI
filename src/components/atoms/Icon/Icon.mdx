---
name: Icon
menu: Atoms
---

import { Props, Playground } from "docz";
import { Icon } from "./Icon.tsx";

# Icon

A Icon is a clickable view to carry out actions such as clearing user input or submitting the form.

# Usage

```tsx
import { Icon } from "fixit-mobile-components";

<Icon />;
```

## Props

<Props of={Icon}></Props>

## Default

<Playground>
  <Icon name="emoticon-kiss" />
</Playground>

## Size

The default size is 24.

<Playground>
  <Icon name="format-size" size={10} />
  <Icon name="format-size" size={20} />
  <Icon name="format-size" size={30} />
  <Icon name="format-size" size={40} />
  <Icon name="format-size" size={50} />
  <Icon name="format-size" size={60} />
  <Icon name="format-size" size={70} />
</Playground>

## Color

The colors must be from the Fixit color pallette.

<Playground>
  <Icon name="palette-swatch" color="primary" />
  <Icon name="palette-swatch" color="accent" />
  <Icon name="palette-swatch" color="grey" />
  <Icon name="palette-swatch" color="light" />
  <Icon name="palette-swatch" color="dark" />
  <Icon name="palette-swatch" color="red" />
  <Icon name="palette-swatch" color="orange" />
  <Icon name="palette-swatch" color="yellow" />
  <Icon name="palette-swatch" color="blue" />
  <Icon name="palette-swatch" color="green" />
  <Icon name="palette-swatch" color="notification" />
  <Icon name="palette-swatch" color="error" />
  <Icon name="palette-swatch" color="success" />
</Playground>

## Icon Library

<Playground>
  <Icon library="Entypo" name="github" />
</Playground>
