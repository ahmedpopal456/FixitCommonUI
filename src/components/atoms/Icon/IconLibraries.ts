import AntDesign from 'react-native-vector-icons/AntDesign';
import Entypo from 'react-native-vector-icons/Entypo';
import EvilIcons from 'react-native-vector-icons/EvilIcons';
import Feather from 'react-native-vector-icons/Feather';
import FontAwesome from 'react-native-vector-icons/FontAwesome';
import FontAwesome5 from 'react-native-vector-icons/FontAwesome5';
import Fontisto from 'react-native-vector-icons/Fontisto';
import Foundation from 'react-native-vector-icons/Foundation';
import Ionicons from 'react-native-vector-icons/Ionicons';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import Octicons from 'react-native-vector-icons/Octicons';
import Zocial from 'react-native-vector-icons/Zocial';
import SimpleLineIcons from 'react-native-vector-icons/SimpleLineIcons';

export const IconLibraries = {
  AntDesign,
  Entypo,
  EvilIcons,
  Feather,
  FontAwesome,
  FontAwesome5,
  Fontisto,
  Foundation,
  Ionicons,
  MaterialIcons,
  MaterialCommunityIcons,
  Octicons,
  Zocial,
  SimpleLineIcons,
};
