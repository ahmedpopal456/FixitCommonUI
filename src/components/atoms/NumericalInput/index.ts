import { NumericalInput } from './NumericalInput';

export { NumericalInput };
