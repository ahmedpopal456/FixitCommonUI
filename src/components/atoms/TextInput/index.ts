import { TextBox } from './TextInput';

export { TextBox };
