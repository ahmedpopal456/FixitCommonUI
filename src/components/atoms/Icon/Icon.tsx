import React from "react";

import colors from "../../../theme/colors";

import { IconLibraries } from "./IconLibraries";
import { IconProps } from "./icon-props.model";

/** Building the Icon */
export const Icon: React.FC<IconProps> = (props) => {
  const VectorIcon = IconLibraries[props.library!];
  const color: string = colors[props.color!];

  const childProps = { ...props };
  delete childProps.color;

  return <VectorIcon {...childProps} name={props.name} color={color} />;
};

Icon.defaultProps = {
  library: "MaterialCommunityIcons",
  color: "primary",
  size: 24,
};

export default Icon;
