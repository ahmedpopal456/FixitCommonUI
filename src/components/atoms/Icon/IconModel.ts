import { TextStyle } from 'react-native';
import colors from '../../../theme/colors';

import { IconLibraries } from './IconLibraries';

export interface IconProps {
  style?: TextStyle & React.CSSProperties;
  /** Any color from the theme's color palette */
  color?: keyof typeof colors;
  /** Size of the icon */
  size?: number;
  /**  The library which the icon is from */
  library?: keyof typeof IconLibraries;
  /**  The name of the icon */
  name: string;
}
