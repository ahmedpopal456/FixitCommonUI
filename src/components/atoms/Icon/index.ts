import { Icon } from './Icon';
import { IconLibraries } from './IconLibraries';

export { Icon, IconLibraries };
