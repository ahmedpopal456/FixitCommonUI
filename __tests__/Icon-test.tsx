/**
 * @format
 */

import 'react-native';
import React from 'react';
import { Icon } from '../src/components/atoms/Icon';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders name', () => {
  renderer.create(<Icon name="home" />);
});

it('renders library', () => {
  renderer.create(<Icon library="AntDesign" name="home" />);
});

it('renders color', () => {
  renderer.create(<Icon library="AntDesign" name="home" color="accent"/>);
});

it("matches snapshot", () => {
  const button = renderer
    .create(<Icon library="AntDesign" name="home" color="accent"/>)
    .toJSON();
  expect(button).toMatchSnapshot();
});

