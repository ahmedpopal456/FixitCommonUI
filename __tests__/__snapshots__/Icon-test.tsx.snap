// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches color snapshot 1`] = `
<Text
  allowFontScaling={false}
  library="AntDesign"
  style={
    Array [
      Object {
        "color": "#FFD14A",
        "fontSize": 24,
      },
      undefined,
      Object {
        "fontFamily": "anticon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;

exports[`matches library snapshot 1`] = `
<Text
  allowFontScaling={false}
  library="AntDesign"
  style={
    Array [
      Object {
        "color": "#33343E",
        "fontSize": 24,
      },
      undefined,
      Object {
        "fontFamily": "anticon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;

exports[`matches name snapshot 1`] = `
<Text
  allowFontScaling={false}
  library="MaterialCommunityIcons"
  style={
    Array [
      Object {
        "color": "#33343E",
        "fontSize": 24,
      },
      undefined,
      Object {
        "fontFamily": "Material Design Icons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  󰋜
</Text>
`;
